web-console/src/components/table-column-selection.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Button, Checkbox, Popover, Position } from "@blueprintjs/core";
+import * as React from 'react';
+
+import { FormGroup, IconNames } from "./filler";
+
+import "./table-column-selection.scss";
+
+interface TableColumnSelectionProps extends React.Props<any> {
+  tableName: string;
+  onChange: () => void;
+}
+
+interface TableColumnSelectionState {
+  tableColumnSelection: any;
+}
+
+export class TableColumnSelection extends React.Component<TableColumnSelectionProps, TableColumnSelectionState> {
+
+  constructor(props: TableColumnSelectionProps) {
+    super(props);
+    this.state = {
+      tableColumnSelection: JSON.parse(localStorage[this.props.tableName])
```

Comment:
There are two issues here:

1. You should use `localStorageSet` and `localStorageGet` functions in the utils (have a look at what they do)
2. The JSON.parse should be wrapped in a reasonable try catch since right now if `localStorage[this.props.tableName]` returns invalid JSON this will crash. You sort of luck out here because on initial run `localStorage[this.props.tableName]` will return `null` which JSON.parse will String to `"null"` and then parse as `null`